alpaca/data/models/auctions.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+from typing import Dict, List
+
+from alpaca.common.models import ValidateBaseModel as BaseModel
+from alpaca.common.types import RawData
+from alpaca.common.enums import Sort
+from alpaca.data.mappings import AUCTION_MAPPING
+from alpaca.data.models.base import BaseDataSet, TimeSeriesMixin
+
+
+class Auction(BaseModel):
+    """Represents one auction of aggregated trade data over a specified interval.
+
+    Attributes:
+        symbol (str): The ticker identifier for the security whose data forms the bar.
+        timestamp (datetime): The timestamp of the auction.
+        condition (str): The condition of the auction.
+        price (float): The price of the auction.
+        size (float): The size of the auction.
+        exchange (str): The exchange of the auction.
+        auction_type (str): The type of auction. OPEN or CLOSE.
+
+    """
+
+    symbol: str
+    timestamp: datetime
+    condition: str
+    price: float
+    size: float
+    exchange: str
+    auction_type: str
+
+    def __init__(self, symbol: str, raw_data: RawData) -> None:
+        """Instantiates an auction
+
+        Args:
+            raw_data (RawData): Raw unparsed auction data from API, contains ohlc and other fields.
+        """
+        mapped_auction = {}
+
+        if raw_data is not None:
+            mapped_auction = {
+                AUCTION_MAPPING[key]: val
+                for key, val in raw_data.items()
+                if key in AUCTION_MAPPING
+            }
+
+        super().__init__(symbol=symbol, **mapped_auction)
+
+
+class AuctionSet(BaseDataSet, TimeSeriesMixin):
+    """A collection of Auctions.
+
+    Attributes:
+        data (Dict[str, List[Auction]]): The collection of Auctions keyed by symbol.
+    """
+
+    data: Dict[str, List[Auction]] = {}
+
+    def __init__(self, raw_data: RawData, sort: Sort) -> None:
```

Comment:
StockAuctionsRequest.sort is `Optional[Sort]`

```suggestion
    def __init__(self, raw_data: RawData, sort: Optional[Sort]) -> None:
```